go/arrow/array/map_test.go

```diff
@@ -0,0 +1,151 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package array_test
+
+import (
+	"testing"
+
+	"github.com/apache/arrow/go/arrow"
+	"github.com/apache/arrow/go/arrow/array"
+	"github.com/apache/arrow/go/arrow/memory"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMapArray(t *testing.T) {
+	pool := memory.NewCheckedAllocator(memory.NewGoAllocator())
+	defer pool.AssertSize(t, 0)
+
+	var (
+		arr, equalArr, unequalArr *array.Map
+
+		equalValid     = []bool{true, true, true, true, true, true, true}
+		equalOffsets   = []int32{0, 1, 2, 5, 6, 7, 8, 10}
+		equalKeys      = []string{"a", "a", "a", "b", "c", "a", "a", "a", "a", "b"}
+		equalValues    = []int32{1, 2, 3, 4, 5, 2, 2, 2, 5, 6}
+		unequalValid   = []bool{true, true, true}
+		unequalOffsets = []int32{0, 1, 4, 7}
+		unequalKeys    = []string{"a", "a", "b", "c", "a", "b", "c"}
+		unequalValues  = []int32{1, 2, 2, 2, 3, 4, 5}
+	)
+
+	bldr := array.NewMapBuilder(pool, arrow.BinaryTypes.String, arrow.PrimitiveTypes.Int32, false)
+	defer bldr.Release()
+
+	kb := bldr.KeyBuilder().(*array.StringBuilder)
+	ib := bldr.ItemBuilder().(*array.Int32Builder)
+
+	bldr.AppendValues(equalOffsets, equalValid)
+	for _, k := range equalKeys {
+		kb.Append(k)
+	}
+	ib.AppendValues(equalValues, nil)
+
+	assert.Equal(t, len(equalValid), bldr.Len())
+	assert.Zero(t, bldr.NullN())
+
+	arr = bldr.NewMapArray()
+	defer arr.Release()
+
+	bldr.AppendValues(equalOffsets, equalValid)
+	for _, k := range equalKeys {
+		kb.Append(k)
+	}
+	ib.AppendValues(equalValues, nil)
+
+	equalArr = bldr.NewMapArray()
+	defer equalArr.Release()
+
+	bldr.AppendValues(unequalOffsets, unequalValid)
+	for _, k := range unequalKeys {
+		kb.Append(k)
+	}
+	ib.AppendValues(unequalValues, nil)
+
+	unequalArr = bldr.NewMapArray()
+	defer unequalArr.Release()
+
+	assert.True(t, array.ArrayEqual(arr, arr))
+	assert.True(t, array.ArrayEqual(arr, equalArr))
+	assert.True(t, array.ArrayEqual(equalArr, arr))
+	assert.False(t, array.ArrayEqual(equalArr, unequalArr))
+	assert.False(t, array.ArrayEqual(unequalArr, equalArr))
+
+	// assert.True(t, array.ArraySliceEqual(arr, 0, 1, unequalArr, 0, 1))
```

Comment:
why are these commented out?

Reply:
forgot to uncomment when testing something. :) Uncommented now.